app/models/viewmodels/notificationspage/MovementTotalsResponseConverter.scala

```diff
@@ -39,8 +39,10 @@ class MovementTotalsResponseConverter @Inject()(decoder: Decoder, dateTimeFormat
     if (canConvertFrom(notification)) {
 
       val crcCodeExplanation = notification.crcCode.flatMap(buildCrcCodeExplanation)
-      val roeCodeExplanation = notification.masterRoe.flatMap(buildRoeCodeExplanation)
-      val soeCodeExplanation = notification.masterSoe.flatMap(buildSoeCodeExplanation)
+      val roeCodeExplanation =
+        notification.entries.headOption.flatMap(_.entryStatus).flatMap(_.roe).flatMap(buildRoeCodeExplanation)
```

Comment:
Looks like there is some logic connected to that `Seq` - it would be nice to have `Entries` object which would encapsulate logic related this Seq to separate object.

This is concept called [first class collection](https://softwareengineering.stackexchange.com/questions/139353/why-should-we-preferably-use-first-class-collections) which brings much more